actions/get-vault-secrets/action.yaml

```diff
@@ -0,0 +1,46 @@
+name: Get Vault Secrets
+description: Composite action (step) to get secrets from Grafana Labs' Vault instance.
+inputs:
+  workload_identity_provider:
+    required: true
+    description: Set to `secrets.WORKLOAD_IDENTITY_POOL_PROVIDER`
+  iap_service_account:
+    required: true
+    description: Set to `secrets.VAULT_IAP_SA_EMAIL`
+  iap_audience:
+    required: true
+    description: Set to `secrets.VAULT_IAP_OAUTH_CLIENT_ID`
+  vault_url:
+    required: true
+    description: Set to `secrets.VAULT_URL`
+  secrets:
+    required: true
+    description: Secret mapping. See https://github.com/hashicorp/vault-action for the format.
```

Comment:
Not sure. I think the full path is nice because:
- it's easily understandable by (new) users
- you can, for example, do code search for `ci/data/common/my-secret` (not just `my-secret`) to get all usages of that secret without any conflicts
- There is no conflicting formats (one by us and one documented by Hashicorp that users can be confused by

Reply:
I find the full Vault format obscure and confusing to people - for example, it's common for `data` to get forgotten or added unnecessarily.

This stream of work gives us a good chance to think about the interfaces we offer to people and what we can do to make their lives easier. We can do better than passing through what Vault offers directly.

You don't have to take what I suggested literally if you don't like it, but let's explore what could be done.

Reply:
New format:
```
      - id: get-secrets
        uses: grafana/shared-workflows/actions/get-vault-secrets@main
        with:
          # Secrets placed in the ci/common/<path> path in Vault
          common_secrets: |
            ENVVAR1=test-secret:testing
          # Secrets placed in the ci/repo/<repo>/<path> path in Vault
          repo_secrets: |
            ENVVAR2=test-secret:key1
```

Let me know what you think